### DISCUSSION TITLE
Help ;3

### DISCUSSION THREAD

**Discussion by staaandup** [SteamDatabase/ValveResourceFormat#415] [General] [answered]
### Describe the bug

Hey, guys. Sorry for bothering you, but I can't join the discord because my steam account is restricted, and I was hoping you could help me with a simple question.
I'd like to extract specific files from Dota 2's "pak01 dir" using the command line "cmd" and the program VRF.exe but it doesn't appear to be working. I must be doing something wrong.
This is what I typed into cmd, and when I press Enter, vrf.exe opens the "pak01 dir" but doesn't extract anything.

VRF "D:\Steam\steamapps\common\dota 2 beta\game\dota\pak01_dir.vpk" "D:\SomeFolder\particles\dire_fx\dire_lava_falling_rocks_sparks.vpcf" "D:\SomeFolder\particles\dire_fx\dire_lfr_sparks_13sec.vpcf" "D:\SomeFolder\particles\dire_fx\dire_lfr_sparks_19sec.vpcf"

What I hope to gain from this is the ability to use cmd and vrf.exe to automatically extract thousands of SPECIFIC files to a SPECIFIC folder.

### What Valve game does this happen in

Dota 2

### Reference file in a Source 2 game

.

### Expected behavior

.

### What version of VRF are you using? On what platform?

.

**Comment by xPaw** [comment, marked as answer]
Use decompiler.exe

VRF.exe is a GUI application.

**Comment by staaandup** [comment]
Thank you very much! and apologies But, if you don't mind, could you please tell me whats the correct "cmd" command I should use with decompiler.exe to extract three specific files from Dota 2's "pak01 dir"? I've tried this and a few others, but I'm a complete noob at this and it's not working.

decompiler.exe "D:\Steam\steamapps\common\dota 2 beta\game\dota\pak01_dir.vpk" "D:\SomeFolder\particles\dire_fx\dire_lava_falling_rocks_sparks.vpcf" "D:\SomeFolder\particles\dire_fx\dire_lfr_sparks_13sec.vpcf" "D:\SomeFolder\particles\dire_fx\dire_lfr_sparks_19sec.vpcf"

I would appreciate even the smallest example you could provide so that I could use and learn from it.

**Reply by kristiker** [reply]
Ask the program for help first, type `decompiler.exe --help`. You will see some options you can use but it's not guaranteed that it'll have the functionality that you want.  The program should definitely be able to decompile and extract at least **one** file at a time.  Don't know if it can do thousands of files but worst case scenario you'll need to ask it a thousand times.

**Reply by kristiker** [reply]
The filter option `-f` might help.

**Reply by staaandup** [reply]
@kristiker Thank you kindly! You've pointed me in the right direction!

**Comment by staaandup** [comment]
This is the best i've come up with so far, but it's too slow when run in a batch file to extract a large number of specific files.

`Decompiler.exe -i "D:\Steam\steamapps\common\dota 2 beta\game\dota\pak01_dir.vpk" -d -f "particles\addons_gameplay\fire_torch_glow_shadowed.vpcf_c" -o "C:\Output"`
